resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsServerAddressStreamProvider.java

```diff
@@ -66,12 +66,22 @@
         try {
             DirContext ctx = new InitialDirContext(env);
             String dnsUrls = (String) ctx.getEnvironment().get("java.naming.provider.url");
-            String[] servers = dnsUrls.split(" ");
-            for (String server : servers) {
-                try {
-                    defaultNameServers.add(SocketUtils.socketAddress(new URI(server).getHost(), DNS_PORT));
-                } catch (URISyntaxException e) {
-                    logger.debug("Skipping a malformed nameserver URI: {}", server, e);
+            // Only try if not empty as otherwise we will produce an exception
+            if (dnsUrls != null && !dnsUrls.isEmpty()) {
+                String[] servers = dnsUrls.split(" ");
+                for (String server : servers) {
+                    try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1141)